src/ol/layer/Graticule.js

```diff
@@ -379,6 +378,7 @@ class Graticule extends VectorLayer {
                 })
               })
           });
+          this.latLabelStyleCache_[label].getText().setText(label);
```

Comment:
With this addition, it is no longer necessary to cache or clone the style.

You could remove the `lonLabelStyleCache_` and `latLabelStyleCache_` members entirely, and create `lonLabelStyle_` and `latLabelStyle_` members instead, which either get the provided `options.lonLabelStyle` and `options.latLabelStyle` assigned, or the default that's created when these are not provided.

Reply:
Yes, I had the same thought but concentrated on the minimum change to fix the bug.  I have kept a single clone of each style we are updating to guard against the possibility of a user using the supplied definition elsewhere.